### PR TITLE
Created lua-check.yml file

### PR DESCRIPTION
**Pull request by shabbir23ah**
GitHub action .yml file for Lua code formatting and linting according issue  #4433

### Pull Request Prelude

- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

Add Luacheck github action #4433 

**Issues addressed:** 

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

### CODE REVIEW

File: .github/workflows/lua-check.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Lua Lint and Format
+
+on:
+  push:
+    branches:
+      - master
+      - v*
+    paths:
+      - 'src/**.lua'
```
Comment (marmichalski) [inline .github/workflows/lua-check.yml:9]:
```suggestion
      - 'data/**.lua'
      - 'config.lua.dist'
```

File: .github/workflows/lua-check.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  pull_request:
+    paths:
+      - 'src/**.lua'
```
Comment (marmichalski) [inline .github/workflows/lua-check.yml:12]:
```suggestion
      - 'data/**.lua'
      - 'config.lua.dist'
```

File: .github/workflows/lua-check.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          luaVersion: "5.3"
+
+      - name: Install Luacheck
+        run: |
+          sudo apt-get install lua-filesystem
+          sudo luarocks install luacheck
+
+      - name: Run Luacheck
+        run: luacheck .
```
Comment (marmichalski) [inline .github/workflows/lua-check.yml:31]:
```suggestion
        run: luacheck data config.lua.dist
```

File: .github/workflows/lua-check.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          luaVersion: "5.3"
+
+      - name: Install LuaFormatter
+        run: |
+          sudo luarocks install luaformatter
+
+      - name: Run LuaFormatter
+        run: |
+          find . -name "*.lua" -exec luaformatter -i {} \;
```
Comment (marmichalski) [inline .github/workflows/lua-check.yml:49]:
```suggestion
          find . -regex ".*\.lua.*" -exec luaformatter -i {} \;
```

So the config.lua.dist is  also checked :thinking:

**Comment by ranisalt** [comment]
A few suggestions:
- Please add these jobs to the workflow in .github/workflows/lua-syntax.yml so that related checks are on the same description
- Use the latest version of Lua and the action (just like in the same file)
- I think renaming it to `lua-check.yml` would be good!

**Comment by shabbir23ah** [comment]
My apologies, allow me to clarify my intentions
I will combine lua-check.yml jobs into lua-syntex.yml to keep all jobs in single file and rename the file's name to lua-check.yml, right?

**Comment by ranisalt** [comment]
> I will combine lua-check.yml jobs into lua-syntex.yml to keep all jobs in single file and rename the file's name to lua-check.yml, right?

Yes, precisely

**Comment by shabbir23ah** [comment]
consolidate all jobs into single file` lua-check.yml` and updated version

**Review by DSpeichert — APPROVED** [review]

**Review by EPuncker — APPROVED** [review]

**Pull request merged**